### ISSUE TITLE
Active attention

### ISSUE THREAD

**Issue by lemuriandezapada** [fchollet/keras#448]
Hey guys,

Has anyone managed to derive the math in these recent active attention stuff?
http://jmlr.org/proceedings/papers/v37/gregor15.pdf
http://jmlr.org/proceedings/papers/v37/xuc15.pdf

I'm having trouble understanding the concept of how they derive the errors through their new 'layer' and see whether it can actually be implemented in keras.

I think at least the last one a softmax layer and a multiplicative "merge" mode is enough. The first one is rather iffy.

**Comment by vzhong** [comment]
I think you'll need a nested scan inside the recurrence to calculate scores for the softmax at each timestep, as the input is of variable length. Then you'll need to do a multiplicative merge to combine the features.

**Comment by lemuriandezapada** [comment]
The multiplicative merge is easy. The nested scan sounds a bit tricky as it exceeds my theano skills. 
I think I also need to change the graph model to allow as inputs layers that haven't been declared yet, if I understand correctly.

**Comment by pranv** [comment]
See https://github.com/fchollet/keras/issues/443
